authority/admin/api/acme.go

```diff
@@ -88,159 +85,31 @@ func (h *Handler) provisionerHasEABEnabled(ctx context.Context, provisionerName
-		return
-	}
-
-	ctx := r.Context()
-	prov, err := provisionerFromContext(ctx)
-	if err != nil {
-		api.WriteError(w, admin.WrapErrorISE(err, "error getting provisioner from context"))
-		return
-	}
-
-	// check if a key with the reference does not exist (only when a reference was in the request)
-	reference := body.Reference
-	if reference != "" {
-		k, err := h.acmeDB.GetExternalAccountKeyByReference(ctx, prov.GetId(), reference)
-		// retrieving an EAB key from DB results in an error if it doesn't exist, which is what we're looking for,
-		// but other errors can also happen. Return early if that happens; continuing if it was acme.ErrNotFound.
-		if shouldWriteError := err != nil && !errors.Is(err, acme.ErrNotFound); shouldWriteError {
-			api.WriteError(w, admin.WrapErrorISE(err, "could not lookup external account key by reference"))
-			return
-		}
-		// if a key was found, return HTTP 409 conflict
-		if k != nil {
-			err := admin.NewError(admin.ErrorBadRequestType, "an ACME EAB key for provisioner '%s' with reference '%s' already exists", prov.GetName(), reference)
-			err.Status = 409
-			api.WriteError(w, err)
-			return
-		}
-		// continue execution if no key was found for the reference
-	}
-
-	eak, err := h.acmeDB.CreateExternalAccountKey(ctx, prov.GetId(), reference)
-	if err != nil {
-		msg := fmt.Sprintf("error creating ACME EAB key for provisioner '%s'", prov.GetName())
-		if reference != "" {
-			msg += fmt.Sprintf(" and reference '%s'", reference)
-		}
-		api.WriteError(w, admin.WrapErrorISE(err, msg))
-		return
-	}
-
-	response := &linkedca.EABKey{
-		Id:          eak.ID,
-		HmacKey:     eak.KeyBytes,
-		Provisioner: prov.GetName(),
-		Reference:   eak.Reference,
-	}
+// ACMEAdminResponder is responsible for writing ACME admin responses
+type ACMEAdminResponder struct{}
 
-	api.ProtoJSONStatus(w, response, http.StatusCreated)
+// NewACMEAdminResponder returns a new ACMEAdminResponder
+func NewACMEAdminResponder() *ACMEAdminResponder {
+	return &ACMEAdminResponder{}
 }
 
-// DeleteExternalAccountKey deletes an ACME External Account Key.
-func (h *Handler) DeleteExternalAccountKey(w http.ResponseWriter, r *http.Request) {
-
-	keyID := chi.URLParam(r, "id")
-
-	ctx := r.Context()
-	prov, err := provisionerFromContext(ctx)
-	if err != nil {
-		api.WriteError(w, admin.WrapErrorISE(err, "error getting provisioner from context"))
-		return
-	}
-
-	if err := h.acmeDB.DeleteExternalAccountKey(ctx, prov.GetId(), keyID); err != nil {
-		api.WriteError(w, admin.WrapErrorISE(err, "error deleting ACME EAB Key '%s'", keyID))
-		return
-	}
-
-	api.JSON(w, &DeleteResponse{Status: "ok"})
+// GetExternalAccountKeys writes the response for the EAB keys GET endpoint
+func (h *ACMEAdminResponder) GetExternalAccountKeys(w http.ResponseWriter, r *http.Request) {
+	api.WriteError(w, admin.NewError(admin.ErrorNotImplementedType, "this functionality is currently only available in Certificate Manager: https://smallstep.com/signup?product=cm"))
```

Comment:
@dopey: do you think this is sufficient, or does it need more oomph?

I tried to make this a clickable link (e.g. something akin to https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda), but that might not always work as expected. This is just a plain text link to be on the safe side.